packages/backend-function/src/factory.ts

```
@@ -294,30 +285,36 @@ const nodeVersionMap: Record<NodeVersion, Runtime> = {
   20: Runtime.NODEJS_20_X,
 };
 
-const secretPathSuffix = '_PATH';
-const secretPlaceholderText = '<value will be resolved during runtime>';
 const secretPaths: string[] = [];
 
 const translateEnvironmentProp = (
   functionEnvironmentProp: HydratedFunctionProps['environment'],
   backendSecretResolver: BackendSecretResolver
 ): Record<string, string> => {
+  const secretPlaceholderText = '<value will be resolved during runtime>';
+  const amplifySecretPaths = 'AMPLIFY_SECRET_PATHS';
+  const secretPathEnvVars: Record<string, string> = {};
   const result: Record<string, string> = {};
-  const secretPathEnvVars = [];
 
   for (const [key, value] of Object.entries(functionEnvironmentProp)) {
     if (typeof value !== 'string') {
-      const secretPath = backendSecretResolver.resolveToPath(value);
-      result[key + secretPathSuffix] = secretPath;
+      const secretPath = backendSecretResolver.resolvePath(value);
       result[key] = secretPlaceholderText;
-      secretPathEnvVars.push(key + secretPathSuffix);
+      secretPathEnvVars[key] = secretPath;
       secretPaths.push(secretPath);
     } else {
+      if (key === amplifySecretPaths) {
```

We should throw this error regardless of the value type